### PR TITLE
Importing Resource Container from Zip File

### PR DESCRIPTION
**Pull request by KJoslyn**
Dependent on PR in kotlin-resource-container

### CODE REVIEW

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/menu/view/MainMenu.kt [hunk 1]
```diff
@@ -34,7 +35,14 @@ class MainMenu : MenuBar() {
                         Platform.runLater { if (it) dialog.open() else dialog.close() }
                     }
                     action {
-                        val file = chooseDirectory(messages["importResourceTip"])
+//                        val file = chooseDirectory(messages["importResourceTip"])
+//                        file?.let {
+//                            viewModel.importContainerDirectory(file)
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/menu/view/MainMenu.kt]: We should still allow directory imports, right?

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/menu/view/MainMenu.kt [hunk 2]
```diff
@@ -19,27 +20,41 @@ class MainMenu : MenuBar() {
...
+                            file?.let {
+                                viewModel.importContainerDirectory(file)
+                            }
+                        }
+                importMenuItem(messages["importResourceFromZip"])
+                        .setOnAction {
+                            val file = chooseFile(
+                                    messages["importResourceFromZip"],
+                                    arrayOf(FileChooser.ExtensionFilter("Zip files (*.zip)", "*.zip")),
+                                    FileChooserMode.Single).firstOrNull()
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/menu/view/MainMenu.kt]:
```
    FileChooserMode.Single
).firstOrNull()
```

File: .travis.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ cache:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper
 install:
-  - git clone --branch=master https://github.com/WycliffeAssociates/kotlin-resource-container.git ../kotlin-resource-container
+  - git clone --branch=dev https://github.com/WycliffeAssociates/kotlin-resource-container.git ../kotlin-resource-container
```
Comment (danparisd) [inline .travis.yml:12]: thumbs up

File: .travis.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ cache:
...
   - git clone https://github.com/WycliffeAssociates/8woc2018-common.git ../8woc2018-common
```
Comment (aunger) [inline .travis.yml:13]: Maybe you should edit `init_common.sh` (and rename it to `init_sub_projects.sh`), so Travis can build in-flight features when kotlin-resource-container hasn't yet been merged.

...but I don't know whether any of this should happen in a zip file PR.

@sarabiaj might correct me on this idea though....

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/domain/resourceontainer/project/ZipEntryTreeBuilder.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package org.wycliffeassociates.otter.jvm.domain.resourceontainer.project
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/domain/resourceontainer/project/ZipEntryTreeBuilder.kt]: Misspelled ontainer

File: .travis.yml (hunk 4 above)
Comment (jsarabia) [reply .travis.yml:13]: This seems reasonable

I'm also open to revisiting the idea of putting all modules in one repo as well though (and adding into it Android too)

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/CollectionRepository.kt [hunk 6]
```diff
@@ -273,27 +276,28 @@ class CollectionRepository(
...
+                                            } else {
+                                                projectEntity.sort
+                                            }
+                                            identifier = projectEntity.slug
+                                            path = "./${projectEntity.slug}"
+                                            // This title will not be localized into the target language
+                                            title = projectEntity.title
+                                            // Unable to get these fields from the source collection
+                                            categories = listOf()
+                                            versification = ""
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/CollectionRepository.kt]: aren't these the default values for categories and versification anyway?

File: .travis.yml (hunk 4 above)
Comment (aunger) [reply .travis.yml:13]: @sarabiaj will talk to @danparisd about whether we need something different here, so this is resolved for the sake of this PR.

**Review by aunger — APPROVED** [review]

**Pull request merged**